core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2332,12 +2330,24 @@ String memberId() {
             return memberId;
         }
 
+        private RecordState rollbackState() {
+            if (rollbackState == null) {
+                // This case could occur when the batch/offset hasn't transitioned even once or the state transitions have
+                // been committed.
+                return null;
+            }
+            return rollbackState.state;
+        }
+
         // Visible for testing.
         TimerTask acquisitionLockTimeoutTask() {
             return acquisitionLockTimeoutTask;
         }
 
-        void updateAcquisitionLockTimeoutTask(AcquisitionLockTimerTask acquisitionLockTimeoutTask) {
+        void updateAcquisitionLockTimeoutTask(AcquisitionLockTimerTask acquisitionLockTimeoutTask) throws IllegalStateException {
+            if (this.acquisitionLockTimeoutTask != null) {
+                throw new IllegalStateException("The acquisition lock timeout task is being overridden");
```

Comment:
`"The acquisition lock timeout task is being overridden"` I didn't understanf the error message on `null` `acquisitionLockTimeoutTask`

Reply:
quotig @chia7712's comment https://github.com/apache/kafka/pull/17965?new_mergebox=false#pullrequestreview-2476424111 

> One of the issues is the leakage of acquisitionLockTimeoutTask. Therefore, I recommend adding a check to updateAcquisitionLockTimeoutTask that throws an exception if there is an attempt to override an existing updateAcquisitionLockTimeoutTask.

So, basically if you're trying to override a non null `acquisitionLockTimeoutTask`, you will get this exception, hence there won't be any chances of leakage of `acquisitionLockTimeoutTask`. If you feel the error message can be enhanced, feel free to suggest some. I thought this message is intutive.

Reply:
My question was on the error message, I didn't get that.

Reply:
enhanced the error message